pkg/pillar/devicenetwork/dnc.go

```diff
@@ -228,31 +228,36 @@ func VerifyPending(ctx *DeviceNetworkContext, pending *DPCPending,
 	log.Functionf("VerifyPending: No required ports held in pciBack. " +
 		"parsing device port config list")
 
-	ifname, err := checkInterfacesExists(log, pending.PendDPC)
-	if err != nil {
+	portErrors, runnableDPC := checkInterfacesExists(log, pending.PendDPC)
+	if len(portErrors) > 0 {
 		// Still waiting for a network interface to appear
 		if pending.TestCount < MaxDPCCheckIfCount {
-			log.Warnf("VerifyPending: interface check: retry due to ifname %s at count %d: %s",
-				ifname, pending.TestCount, err)
+			log.Warnf("VerifyPending: interface check: retry due to %d port Errors at test count %d",
+				len(portErrors), pending.TestCount)
 			pending.TestCount++
 			return types.DPC_INTF_WAIT
 		}
-		log.Warnf("VerifyPending: interface check: failed due to ifname %s: %s",
-			ifname, err)
-		pending.PendDPC.RecordPortFailure(ifname, err.Error())
-		pending.PendDPC.RecordFailure(err.Error())
+		for _, portError := range portErrors {
+			log.Warnf("VerifyPending: interface check: failed due to ifname %s: %s",
+				portError.ifName, portError.err)
+			pending.PendDPC.RecordPortFailure(portError.ifName, portError.err.Error())
+			pending.PendDPC.RecordFailure(portError.err.Error())
+		}
 		// Proceed trying other interfaces
+		log.Warnf("VerifyPending: Some required ports are missing. Continuing verification process with remaining ports")
```

Comment:
Wouldn't it be useful to log the remaing ports as part of this?

Reply:
Missing port and associated errors are already printed as port of the log.Warn in loop above. This log is basically a brief description/hint that we are not running on all cylinders.